adapters/telaria/telaria.go

```diff
@@ -0,0 +1,331 @@
+func (a *TelariaAdapter) CheckHasImps(request *openrtb.BidRequest) error {
+	if len(request.Imp) == 0 {
+		err := &errortypes.BadInput{
+			Message: "No imp object in the bid request",
+		}
+		return err
+	}
+	return nil
+}
+
+// Checking if Imp[i].Video exists and Imp[i].Banner doesn't exist
+func (a *TelariaAdapter) CheckHasVideoObject(request *openrtb.BidRequest) error {
+	hasVideoObject := false
+
+	for _, imp := range request.Imp {
+		if imp.Banner != nil {
+			return &errortypes.BadInput{
+				Message: "Telaria doesn't support banner",
+			}
+		}
+
+		hasVideoObject = hasVideoObject || imp.Video != nil
+	}
+
+	if !hasVideoObject {
+		return &errortypes.BadInput{
+			Message: "No Video object present in Imp object",
+		}
+	}
+
+	return nil
+}
+
+// Fetches the populated header object
+func GetHeaders(request *openrtb.BidRequest) *http.Header {
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+	headers.Add("X-Openrtb-Version", "2.5")
+	headers.Add("Accept-Encoding", "gzip")
+
+	if request.Device != nil {
+		if len(request.Device.UA) > 0 {
+			headers.Add("User-Agent", request.Device.UA)
+		}
+
+		if len(request.Device.IP) > 0 {
+			headers.Add("X-Forwarded-For", request.Device.IP)
+		}
+
+		if len(request.Device.Language) > 0 {
+			headers.Add("Accept-Language", request.Device.Language)
+		}
+
+		if request.Device.DNT != nil {
+			headers.Add("Dnt", strconv.Itoa(int(*request.Device.DNT)))
+		}
+	}
+
+	return &headers
+}
+
+// Checks the imp[i].ext object and returns a imp.ext object as per ExtImpTelaria format
+func (a *TelariaAdapter) FetchTelariaExtImpParams(imp *openrtb.Imp) (*openrtb_ext.ExtImpTelaria, error) {
+	var bidderExt adapters.ExtImpBidder
+	err := json.Unmarshal(imp.Ext, &bidderExt)
+
+	if err != nil {
+		err = &errortypes.BadInput{
+			Message: "ext.bidder not provided",
+		}
+
+		return nil, err
+	}
+
+	var telariaExt openrtb_ext.ExtImpTelaria
+	_ = json.Unmarshal(bidderExt.Bidder, &telariaExt)
+
+	return &telariaExt, nil
+}
+
+// Fetch the id from the appropriate publisher object
+func (a *TelariaAdapter) FetchPublisherId(request *openrtb.BidRequest) string {
+
+	if request.Site != nil {
+		if request.Site.Publisher != nil {
+			if request.Site.Publisher.ID != "" {
+				return request.Site.Publisher.ID
+			}
+		}
+	}
+
+	if request.App != nil {
+		if request.App.Publisher != nil {
+			if request.App.Publisher.ID != "" {
+				return request.App.Publisher.ID
+			}
+		}
+	}
+
+	return ""
+}
+
+// This method changes <site/app>.publisher.id to request.ext.telaria.seatCode
+// And moves the publisher.id to request.ext.originalPublisherId
+func (a *TelariaAdapter) PopulateRequestExtAndPubId(request *openrtb.BidRequest) error {
+	var requestExtIncoming ReqExtIn
+	if err := json.Unmarshal(request.Ext, &requestExtIncoming); err != nil {
+		return err
+	}
+
+	if requestExtIncoming.Telaria.SeatCode == "" {
```

Comment:
Do we expect to have bidder related data in Request extension? We usually have it in impression extension